vms/platformvm/txs/executor/proposal_tx_executor.go

```diff
@@ -320,215 +322,303 @@ func (e *ProposalTxExecutor) RewardValidatorTx(tx *txs.RewardValidatorTx) error
 	if !currentStakerIterator.Next() {
 		return fmt.Errorf("failed to get next staker to remove: %w", database.ErrNotFound)
 	}
-	stakerToRemove := currentStakerIterator.Value()
+	stakerToReward := currentStakerIterator.Value()
 	currentStakerIterator.Release()
 
-	if stakerToRemove.TxID != tx.TxID {
+	if stakerToReward.TxID != tx.TxID {
 		return fmt.Errorf(
 			"%w: %s != %s",
 			ErrRemoveWrongStaker,
-			stakerToRemove.TxID,
+			stakerToReward.TxID,
 			tx.TxID,
 		)
 	}
 
 	// Verify that the chain's timestamp is the validator's end time
 	currentChainTime := e.OnCommitState.GetTimestamp()
-	if !stakerToRemove.EndTime.Equal(currentChainTime) {
+	if !stakerToReward.NextTime.Equal(currentChainTime) {
```

Comment:
NextTime == EndTime for current stakers (the only ones which can be rewarded). Post continuous staking instead NextTime may be different from EndTime.